InterfacePython/TasmanianParticleSwarm.py

```diff
@@ -0,0 +1,295 @@
+        '''
+        pLibDTSG.tsgParticleSwarmState_Destruct(self.pStatePntr)
+
+    def getNumDimensions(self):
+        '''
+        Return the number of dimensions.
+        '''
+        return pLibDTSG.tsgParticleSwarmState_GetNumDimensions(self.pStatePntr)
+
+    def getNumParticles(self):
+        '''
+        Return the number of particles.
+        '''
+        return pLibDTSG.tsgParticleSwarmState_GetNumParticles(self.pStatePntr)
+
+    def getParticlePositions(self):
+        '''
+        Return the particle positions as a 2D NumPy array. The shape of this array is (self.getNumParticles(), self.getNumDimensions())
+        and the i-th row if this array corresponds to the position of the i-th particle.
+        '''
+        iNumDims = self.getNumDimensions()
+        iNumPart = self.getNumParticles()
+        aResult = np.zeros((iNumDims * iNumPart,), np.float64)
+        pLibDTSG.tsgParticleSwarmState_GetParticlePositions(self.pStatePntr, aResult)
+        return aResult.reshape((iNumPart, iNumDims))
+
+    def getParticleVelocities(self):
+        '''
+        Return the particle velocities as a 2D NumPy array. The shape of this array is (self.getNumParticles(), self.getNumDimensions())
+        and the i-th row if this array corresponds to the velocity of the i-th particle.
+        '''
+        iNumDims = self.getNumDimensions()
+        iNumPart = self.getNumParticles()
+        aResult = np.zeros((iNumDims * iNumPart,), np.float64)
+        pLibDTSG.tsgParticleSwarmState_GetParticleVelocities(self.pStatePntr, aResult)
+        return aResult.reshape((iNumPart, iNumDims))
+
+    def getBestParticlePositions(self):
+        '''
+        Return the best particle positions as a 2D NumPy array.  The shape of this array is (self.getNumParticles()+1 , self.getNumDimensions())
+        and the i-th row if this array corresponds to the best position of the i-th particle for the first .getNumParticles()
+        particles. The last row corresponds to the best particle position of the swarm.
+        '''
+        iNumDims = self.getNumDimensions()
+        iNumPart = self.getNumParticles()
+        aResult = np.zeros((iNumDims * (iNumPart + 1),), np.float64)
+        pLibDTSG.tsgParticleSwarmState_GetBestParticlePositions(self.pStatePntr, aResult)
+        return aResult.reshape((iNumPart + 1, iNumDims))
+
+    def getBestPosition(self):
+        '''
+        Return the best particle position of the swarm as a 1D NumPy array. The size of the array is self.getNumDimensions().
+        '''
+        iNumDims = self.getNumDimensions()
+        aResult = np.zeros((iNumDims,), np.float64)
+        pLibDTSG.tsgParticleSwarmState_GetBestPosition(self.pStatePntr, aResult)
+        return aResult
+
+    def isPositionInitialized(self):
+        '''
+        Returns True if the particle positions have been initialized and False otherwise.
+        '''
+        return bool(pLibDTSG.tsgParticleSwarmState_IsPositionInitialized(self.pStatePntr))
```

Comment:
Test this, `bool(0)` will be `False`, but I'm not sure about `bool(c_int(0))` which comes from C++.

If this causes issues, try
```
return (pLibDTSG.tsgParticleSwarmState_IsPositionInitialized(self.pStatePntr) != 0)
```

Reply:
Both `bool(0)` and `bool(c_int(0))` return `False`, while `bool(1)` and `bool(c_int(1))` return `True` as expected.

I also tested these new functions in the sandbox, and they seem to work as expected.

Reply:
OK, I wasn't sure since the documentation wasn't explicit. They talked about `int` but `c_int` can be argued as either int or not-int.